### PR TITLE
Add Version Catalog

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - This PR adds a `libs.versions.toml` file that contains all versions, libraries, and plugins.
 
https://developer.android.com/build/migrate-to-catalogs#groovy

### Checklist

 - [ ] Added a changelog entry
 - [ ] Relevant test coverage

### CODE REVIEW

File: Demo/build.gradle [hunk 1]
```diff
@@ -52,7 +52,7 @@ android {
     }
 
     composeOptions {
-        kotlinCompilerExtensionVersion = "1.4.1"
+        kotlinCompilerExtensionVersion = "1.5.3"
```
Comment (tdchow) [inline Demo/build.gradle:55]: This needed to be updated to match the Kotlin version.

File: Demo/build.gradle [hunk 2]
```diff
@@ -94,34 +93,29 @@ dependencies {
...
-    implementation 'com.google.code.gson:gson:2.10.1'
-    implementation 'de.greenrobot:eventbus:2.4.1'
-    implementation deps.appCompat
-    implementation 'androidx.preference:preference:1.2.1'
-    implementation 'com.google.android.material:material:1.10.0'
-    implementation 'androidx.cardview:cardview:1.0.0'
-    implementation 'androidx.multidex:multidex:2.0.1'
-
-    implementation 'io.card:android-sdk:5.5.1'
-    implementation 'com.braintreepayments:card-form:5.4.0'
```
Comment (tdchow) [inline Demo/build.gradle:106]: These weren't referenced anywhere. Are they safe to remove?

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+gradleNexusPublish = "1.1.0"
+kotlin = "1.9.10"
+coreKtx = "1.12.0"
+androidxLifecycle = "2.6.2"
+androidxRoom = "2.6.1"
+androidxTest = "1.5.0"
+androidxAppcompat = "1.6.0"
+androidxAnnotation = "1.7.0"
+androidxWork = "2.8.1"
+androidxJunit = "1.1.5"
```
Comment (tdchow) [inline gradle/libs.versions.toml:14]: Calling out that versions are camel case, while library and plugin definitions are kabab case. According to the docs, kebab case is used to help with autocomplete in gradle files.

File: settings.gradle [hunk 4]
```diff
@@ -1,3 +1,17 @@
+pluginManagement {
```
Comment (tdchow) [inline settings.gradle:1]: This block is required to be able to resolve plugins in the root build.gradle plugins block.

File: Demo/build.gradle (hunk 2 above)
Comment (jaxdesmarais) [reply Demo/build.gradle:106]: Should be fine to remove. `io.card` is the old snap a photo autofill credit card thing used in the Drop-in and `card-form` is used by the Drop-in as well. Neither should be needed here.

**Review by jaxdesmarais — APPROVED** [review]

File: settings.gradle (hunk 4 above)
Comment (sshropshire) [reply settings.gradle:1]: curious: does gradle sync fail without this block?

Comment (tdchow) [reply settings.gradle:1]: Yep there were failures due to this plugins block in the root build.gradle file:

```groovy
plugins {
    alias libs.plugins.dokka
    alias libs.plugins.gradle.nexus.publish
    alias libs.plugins.detekt

    alias libs.plugins.android.application apply false
    alias libs.plugins.android.library apply false
    alias libs.plugins.kotlin.android apply false
    alias libs.plugins.kotlin.parcelize apply false
    alias libs.plugins.kotlin.kapt apply false
    alias libs.plugins.androidx.safeargs apply false
}
```

File: Demo/build.gradle (hunk 2 above)
Comment (tdchow) [reply Demo/build.gradle:106]: Hmm, I'm now seeing errors when removing `card-form` 🤔 there are references in the demo app. I'll leave it in for now, but it sounds like we could refactor it out at some point.

**Pull request merged**